draft-ietf-tls-esni.md

```diff
@@ -1175,6 +1188,30 @@ as authoritative for the public name, this may trigger the retry logic described
 in {{rejected-ech}} or result in a connection failure. A proxy which is not
 authoritative for the public name cannot forge a signal to disable ECH.
 
+## Deployment Impact {#no-sni}
+
+Some use cases which depend on information ECH encrypts may break with the
+deployment of ECH. The extent of breakage depends on a number of external
+factors, including, for example, whether ECH can be disabled, whether or not
+the party disabling ECH is trusted to do so, and whether or not client
+implementations will fall back to TLS without ECH in the event of disablement.
+
+Depending on implementation details and deployment settings, use cases
+which depend on plaintext TLS information may require fundamentally different
+approaches to continue working. For example, in managed enterprise settings,
+one approach may be to disable ECH entirely via via group policy and for
+client implementations to honor this action. Another approach may be to
+intercept and decrypt client TLS connections. The feasibility of alternative
+solutions is specific to individual deployments.
+
+In environments where the network operator does not control the endpoint devices, or does controls the endpoint devices, but
+is concerned about the security consequences of compromised devices, e.g., data
+exfiltration, the SNI field is unsuitable for use as a control even in the
+absence of ECH. This is because devices without controls, or which have been compromised, can alter or spoof the
+value in an SNI field already, and can even bypass security appliances which
+try to 'double-check' websites hosted by the target server. ECH does not
+materially change this situation.
```

Comment:
```suggestion
In environments where the network operator does not control the endpoint
devices, or does controls the endpoint devices, but is concerned about the
security consequences of compromised devices, e.g., data exfiltration, the
SNI field is unsuitable for use as a control even in the absence of ECH. This
is because devices without controls, or which have been compromised, can alter
or spoof the value in an SNI field already, and can even bypass security
appliances which try to 'double-check' websites hosted by the target server.
ECH does not materially change this situation.
```